### DISCUSSION TITLE
How to properly register code snippets

### DISCUSSION THREAD

**Discussion by snake-py** [microsoft/vscode-discussions#580] [Extension Development QnA] [answered]
Hello I am trying to discover code snippets from multiple sources and then register them. However, right now it seems that `provideCompletionItems` is not executed. I pass this function to `registerCompletionItemProvider`, the registration is executed but not the `provideCompletionItems`. What am I doing wrong? Do I need to call register or so?

```typescript
const registerCompletionItemProvider = (
    snippetLanguage: string | string[],
    snippetObject: any,
    name: string
) => {
    vscode.languages.registerCompletionItemProvider(
        snippetLanguage,
        {
            provideCompletionItems(document, position) {
                const start = new vscode.Position(position.line, 0);
                const range = new vscode.Range(start, position);
                const completionItems = Object.keys(snippetObject).map((key) => {
                    const snippet = new vscode.SnippetString(snippetObject[key]);
                    const completionItem = new vscode.CompletionItem(
                        key,
                        vscode.CompletionItemKind.Snippet
                    );
                    completionItem.range = range;
                    completionItem.insertText = snippet;
                    return completionItem;
                });
                return completionItems;
            },
        },
        name
    );
};
``` 

Here is the github to the project: https://github.com/snake-py/snippster-vscode-extension/tree/main

**Comment by isidorn** [comment, marked as answer]
Are you still facing this issue with the latest VS COde Insiders?
You might be confusing two concepts, I am not 100% sure what you are tying to achieve :)
Here's a sample that should help you with Snippets https://github.com/microsoft/vscode-extension-samples/tree/main/snippet-sample
And here's one for completion items https://github.com/Microsoft/vscode-extension-samples/tree/main/completions-sample

**Reply by KangXinzhi** [reply]
hello , i need you help [915](https://github.com/microsoft/vscode-discussions/discussions/915)
